packages/arb-avm-cpp/cmachine/arbcore.go

```diff
@@ -82,13 +82,11 @@ func (ac *ArbCore) MachineMessagesRead() *big.Int {
 
 func (ac *ArbCore) MessagesStatus() (core.MessageStatus, error) {
 	defer runtime.KeepAlive(ac)
+	if err := ac.CheckError(); err != nil {
+		return 0, err
+	}
 	statusRaw := C.arbCoreMessagesStatus(ac.c)
```

Comment:
Is there a possible race condition here where there could be no error before the `arbCoreMessagesStatus` call, but then an error occurs after that check? It might make sense to check for an error again after

Reply:
I figured if that happened, error would just be caught next time.  Moving `CheckError` after the message status check covers all eventualities.